lib/asset/transferasset.js

```diff
@@ -215,6 +215,26 @@ class TransferAsset {
         return data;
     }
 
+    /**
+     * Sets the total amount of time, in milliseconds, that it took for the asset to transfer
+     * in its entirety.
+     *
+     * @param {number} duration Number of milliseconds.
+     */
+    setUploadDuration(duration) {
+        this[PRIVATE].uploadDuration = duration;
+    }
```

Comment:
This should be a `setter` equivalent to the getter below.

Reply:
I would add a setter/getter here to record the transfer start time. Transfer would need to check if a start time is already recorded, if not then set one otherwise ignore it.